tests/system/python/e2e/test_e2e_modbus_c_rtu_pi.py

```diff
@@ -0,0 +1,178 @@
+# -*- coding: utf-8 -*-
+
+# FOGLAMP_BEGIN
+# See: http://foglamp.readthedocs.io/
+# FOGLAMP_END
+
+""" Test system/python/e2e/test_e2e_modbus_c_rtu_pi.py
+
+"""
+
+import http.client
+import json
+import time
+import pytest
+import serial
+import utils
+
+
+__author__ = "Vaibhav Singhal"
+__copyright__ = "Copyright (c) 2019 Dianomic Systems"
+__license__ = "Apache 2.0"
+__version__ = "${VERSION}"
+
+SOUTH_PLUGIN = "modbus-c"
+PLUGIN_NAME = "ModbusC"
+SVC_NAME = "modbus-c"
+ASSET_NAME_1 = "adam4015"
+ASSET_NAME_2 = "adam4017"
+
+
+class TestE2EModbusCMTUPI:
+    def check_connect(self, modbus_serial_port, modbus_baudrate):
+        print("Checking serial port {} at baudrate {}.....".format(modbus_serial_port, modbus_baudrate))
+        try:
+            ser = serial.Serial(modbus_serial_port, modbus_baudrate, timeout=1)
+        except:
+            print("Socket connection failed!!")
+            pytest.skip("Test requires a connected serial port!!")
+
+    def get_ping_status(self, foglamp_url):
+        conn = http.client.HTTPConnection(foglamp_url)
+        conn.request("GET", '/foglamp/ping')
+        r = conn.getresponse()
+        assert 200 == r.status
+        r = r.read().decode()
+        jdoc = json.loads(r)
+        return jdoc
+
+    def get_statistics_map(self, foglamp_url):
+        conn = http.client.HTTPConnection(foglamp_url)
+        conn.request("GET", '/foglamp/statistics')
+        r = conn.getresponse()
+        assert 200 == r.status
+        r = r.read().decode()
+        jdoc = json.loads(r)
+        return utils.serialize_stats_map(jdoc)
+
+    @pytest.fixture
+    def start_south_north(self, reset_and_start_foglamp, add_south, skip_verify_north_interface, remove_directories,
+                          south_branch, foglamp_url, start_north_pi_server_c, pi_host, pi_port, pi_token,
+                          modbus_serial_port, modbus_baudrate):
+        """ This fixture clone a south and north repo and starts both south and north instance
+
+            reset_and_start_foglamp: Fixture that resets and starts foglamp, no explicit invocation, called at start
+            add_south: Fixture that adds a south service with given configuration with enabled or disabled mode
+            remove_directories: Fixture that remove directories created during the tests
+        """
+        self.check_connect(modbus_serial_port, modbus_baudrate)
+        cfg = {"protocol": {"value": "RTU"}, "asset": {"value": "modbus"}, "address": {"value": "127.0.0.1"},
+               "port": {"value": "{}".format("2222")}, "device": {"value": modbus_serial_port},
```

Comment:
address and port should NOT be needed. 

Use the given `--modbus-baudrate` or let it be default (and remove from conftest / arg )